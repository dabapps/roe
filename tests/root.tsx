import * as React from 'react';
import * as renderer from 'react-test-renderer';
import AppRoot, { AppRootUnconnected } from '../src/ts/components/app/root';

describe('AppRoot', () => {

  it('should match snapshot', () => {
    const tree = renderer.create(<AppRoot />);

    expect(tree).toMatchSnapshot();
  });

  it('should accept regular element attributes', () => {
    const tree = renderer.create(<AppRoot className="my-class" />);

    expect(tree).toMatchSnapshot();
  });

  it('should apply classes for fixed nav bar and sticky footer', () => {
    const tree = renderer.create(<AppRootUnconnected hasFixedNavBar hasStickyFooter />);

    expect(tree).toMatchSnapshot();
  });

  it('should apply padding for fixed nav bar and sticky footer', () => {
    const tree = renderer.create(
      <AppRootUnconnected
        hasFixedNavBar
        hasStickyFooter
        navBarHeight={50}
        footerHeight={100}
      />
    );

    expect(tree).toMatchSnapshot();
  });

});
