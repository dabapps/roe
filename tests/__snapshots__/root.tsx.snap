// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoot should accept regular element attributes 1`] = `
<div
  className="app-root my-class"
  style={
    Object {
      "paddingBottom": undefined,
      "paddingTop": undefined,
    }
  }
/>
`;

exports[`AppRoot should apply classes for fixed nav bar and sticky footer 1`] = `
<div
  className="app-root has-sticky-footer has-fixed-nav-bar"
  style={
    Object {
      "paddingBottom": undefined,
      "paddingTop": undefined,
    }
  }
/>
`;

exports[`AppRoot should apply padding for fixed nav bar and sticky footer 1`] = `
<div
  className="app-root has-sticky-footer has-fixed-nav-bar"
  style={
    Object {
      "paddingBottom": 100,
      "paddingTop": 50,
    }
  }
/>
`;

exports[`AppRoot should match snapshot 1`] = `
<div
  className="app-root"
  style={
    Object {
      "paddingBottom": undefined,
      "paddingTop": undefined,
    }
  }
/>
`;
